### PR TITLE
Add tiered storage support for Pulsar SQL

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Add support for Pulsar SQL to be able to query offloaded ledgers

### CODE REVIEW

File: pulsar-sql/presto-pulsar/src/test/java/org/apache/pulsar/sql/presto/TestPulsarConnector.java [hunk 1]
```diff
@@ -625,10 +626,10 @@ public PulsarColumnHandle apply(PulsarInternalColumn pulsarInternalColumn) {
 
             Schema schema = topicsToSchemas.get(topicSchemaName).getType() == SchemaType.AVRO ? AvroSchema.of(SchemaDefinition.<Foo>builder().withPojo(Foo.class).build()) : JSONSchema.of(SchemaDefinition.<Foo>builder().withPojo(Foo.class).build());
 
-            org.apache.pulsar.shade.io.netty.buffer.ByteBuf payload = org.apache.pulsar.shade.io.netty.buffer.Unpooled
+            io.netty.buffer.ByteBuf payload = io.netty.buffer.Unpooled
```
Comment (merlimat) [inline pulsar-sql/presto-pulsar/src/test/java/org/apache/pulsar/sql/presto/TestPulsarConnector.java]: Could we just use `ByteBuf` now?

Comment (jerrypeng) [reply pulsar-sql/presto-pulsar/src/test/java/org/apache/pulsar/sql/presto/TestPulsarConnector.java]: we can

**Comment by jerrypeng** [comment]
rerun java8 tests
rerun integration tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun cpp tests

**Comment by jerrypeng** [comment]
rerun integration tests

**Pull request merged**